presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1482,11 +1485,44 @@ private static String stringColumnEquals(String columnName, String value)
     private static String stringColumnLike(String columnName, String pattern)
     {
         StringBuilder filter = new StringBuilder();
-        filter.append(columnName).append(" LIKE ");
-        quoteStringLiteral(filter, pattern);
+        filter.append(columnName);
+        if (hasWildcards(pattern)) {
+            filter.append(" LIKE ");
+            quoteStringLiteral(filter, pattern);
+        }
+        else {
+            filter.append(" = ");
+            quoteStringLiteral(filter, patternToValue(pattern));
+        }
         return filter.toString();
     }
 
+    private static boolean hasWildcards(String pattern)
+    {
+        boolean escaped = false;
+        for (int i = 0; i < pattern.length(); i++) {
+            if (escaped) {
+                escaped = false;
+                continue;
+            }
+
+            char value = pattern.charAt(i);
+            if (value == SEARCH_ESCAPE) {
+                escaped = true;
+            }
+            else if (value == '%' || value == '_') {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static String patternToValue(String pattern)
+    {
+        checkArgument(!hasWildcards(pattern), "Cannot convert a pattern with wildcards to a literal value");
+        return pattern.replace(SEARCH_ESCAPE + "%", "%").replace(SEARCH_ESCAPE + "_", "_");
```

Comment:
This probably needs to handle an escaped escape character, or the escape character used for other non-pattern values. What do other drivers do here?